index.js

```diff
@@ -297,23 +297,13 @@ module.exports = function (log, indexesPath) {
     }
   }
 
-  const nullBipf = bipf.allocAndEncode(null)
   const undefinedBipf = bipf.allocAndEncode(undefined)
 
   function checkEqual(opData, buffer) {
     const fieldStart = opData.seek(buffer)
 
-    if (opData.value === undefined && fieldStart === -1) return true
-    else if (opData.value === undefined)
-      return bipf.compare(buffer, fieldStart, undefinedBipf, 0) === 0
-    else if (opData.value === null)
-      return bipf.compare(buffer, fieldStart, nullBipf, 0) === 0
-    else if (
-      ~fieldStart &&
-      bipf.compareString(buffer, fieldStart, opData.value) === 0
-    )
-      return true
-    else return false
+    if (fieldStart === -1 && opData.value.equals(undefinedBipf)) return true
```

Comment:
I'm a bit afraid of this, because BIPF (unlike JSON) will encode undefined as actual fields, and maybe one day we might want to `checkEqual` for those undefined fields, but this logic would prevent looking for those fields.

```js
BIPF.allocAndEncode({a: undefined, b: 3})
/*
<Buffer 5d 08 61 0e 02 08 62 22 03 00 00 00>
              |  _____    |     ___________
              a    |      b          |
                 undefined           3
              
*/
```

```
JSON.stringify({a: undefined, b: 3})
// '{"b":3}'
```

How about instead we have a special operator `absent()` which has `op.type == "ABSENT"` and then we implement a `checkAbsent()` function similar to `checkEqual()`?

Reply:
It should work in both cases. If you have something that is json encoded and you put that into bipf, like your example then we will hit this case: `fieldStart === -1 && opData.value.equals(undefinedBipf)` and it will work. On the other hand if you encoded bipf directly, then `a` will be there and be equal to `undefined`. In this case the second bipf.compare should work. So I don't see the need for a new operator?

Reply:
If `fieldStart >= 0` and the opData.value matches undefinedBipf, then it will use `bipf.compare`, yeah.

But I mean that from the perspective of the developer's intention. When they write `equal(seekRoot, undefined, {`, what does the developer want? If they want to find `root` fields that are exactly `undefined`, but instead they get messages where the `root` field doesn't exist, then they are getting results that are not what they were looking for.

Reply:
This is hypothetical, but I'm thinking that it's better to be safe and not creating a gotcha.

Reply:
I think it would be a bigger gotcha with another type. This smells a bit like [3 valued logic](https://en.wikipedia.org/wiki/Three-valued_logic) in SQL where there is this indeterminate third value. As a developer writing something like the isRoot equal function I would expect that `undefined` would just work and I didn't have to think about it. Because I would be putting in a undefined value into the database, so I would expect that I can then query for undefined to get it out again.

Reply:
Okay, we got on a call and agreed to do both things: keep the `undefined` logic as is in this PR, plus introduce `absent()` operator.